CMakeLists.txt

```diff
@@ -156,71 +164,157 @@ if(UMF_BUILD_BENCHMARKS)
 if(UMF_FORMAT_CODE_STYLE)
-    find_program(CLANG_FORMAT NAMES clang-format-15 clang-format-15.0 clang-format)
+    find_program(CLANG_FORMAT NAMES clang-format-15 clang-format-15.0
+                                    clang-format)
+    find_program(CMAKE_FORMAT NAMES cmake-format)
 
     if(CLANG_FORMAT)
         get_program_version_major_minor(${CLANG_FORMAT} CLANG_FORMAT_VERSION)
-        message(STATUS "Found clang-format: ${CLANG_FORMAT} (version: ${CLANG_FORMAT_VERSION})")
+        message(STATUS "Found clang-format: ${CLANG_FORMAT} "
+                       "(version: ${CLANG_FORMAT_VERSION})")
 
+        # Check if clang-format (in correct version) is available for code
+        # formatting.
         set(CLANG_FORMAT_REQUIRED "15.0")
         if(NOT (CLANG_FORMAT_VERSION VERSION_EQUAL CLANG_FORMAT_REQUIRED))
-            message(FATAL_ERROR "Required clang-format version is ${CLANG_FORMAT_REQUIRED}")
+            message(FATAL_ERROR "Required clang-format version is "
+                                "${CLANG_FORMAT_REQUIRED}")
         endif()
-    else()
-        message(FATAL_ERROR "UMF_FORMAT_CODE_STYLE=ON, but clang-format not found (required version: ${CLANG_FORMAT_REQUIRED})")
+    endif()
+
+    if(CMAKE_FORMAT)
+        get_program_version_major_minor(${CMAKE_FORMAT} CMAKE_FORMAT_VERSION)
+        message(STATUS "Found cmake-format: ${CMAKE_FORMAT} "
+                       "(version: ${CMAKE_FORMAT_VERSION})")
+
+        # Check if cmake-format (in correct version) is available for code
+        # formatting.
+        set(CMAKE_FORMAT_REQUIRED "0.6")
+        if(NOT (CMAKE_FORMAT_VERSION VERSION_EQUAL CMAKE_FORMAT_REQUIRED))
+            message(FATAL_ERROR "Required cmake-format version is"
+                                "${CMAKE_FORMAT_REQUIRED}")
+        endif()
+    endif()
+
+    if(NOT CLANG_FORMAT AND NOT CMAKE_FORMAT)
+        message(
+            FATAL_ERROR
+                "UMF_FORMAT_CODE_STYLE=ON, but neither clang-format (required version: "
+                "${CLANG_FORMAT_REQUIRED}) nor cmake-format was found.")
     endif()
 
     # Obtain files for clang-format check
     set(format_glob)
-    foreach(DIR IN ITEMS include src test benchmark)
-        list(APPEND format_glob
+    foreach(
+        DIR IN
+        ITEMS include
+              examples
+              src
+              test
+              benchmark)
+        list(
+            APPEND
+            format_glob
             "${DIR}/*.h"
             "${DIR}/*.hpp"
             "${DIR}/*.c"
             "${DIR}/*.cpp"
             "${DIR}/**/*.h"
             "${DIR}/**/*.hpp"
             "${DIR}/**/*.c"
-            "${DIR}/**/*.cpp"
-        )
+            "${DIR}/**/*.cpp")
     endforeach()
     file(GLOB_RECURSE format_list ${format_glob})
 
-    message(STATUS "Adding clang-format-check and clang-format-apply make targets")
+    message(STATUS "Adding clang-format-check and clang-format-apply make "
```

Comment:
There was a discussion about this in a previous PR, but I want to make sure we agree on this: these targets are not make-specific so better to describe them as CMake targets? @lukaszstolarczuk

Reply:
Yes, that's ok, you can say that it's a make target. This tool is most probably not available on Windows anyway...